sdk/core/core-paging/types/corePaging.d.ts

```diff
@@ -32,5 +32,5 @@ export interface PagedAsyncIterableIterator<T, PageT = never> {
   /**
    * @member {Function} [byPage] Return an AsyncIterableIterator that works a page at a time
    */
-  byPage: (settings?: PageSettings) => AsyncIterableIterator<PageT extends never ? T[] : PageT>;
+  byPage: (settings?: PageSettingsT) => AsyncIterableIterator<PageT extends never ? T[] : PageT>;
```

Comment:
```suggestion
  byPage: (settings?: PageSettingsT) => AsyncIterableIterator<PageT extends never ? T[] : PageT, undefined>;
```

@jeremymeng discovered that the tooling experience of using the manual iterators isn't so good right now. Observe:

```ts
declare var obj: PagedAsyncIterableIterator<T, TPage>;
let record = obj.next();
let value = record.value; // is any :(
```

Setting TReturn to undefined somehow would be ideal, but the above won't work since AsyncIterableIterator only takes a T.

Reply:
Sorry, to be clear, the suggestion is for illustration only, don't take it.

Reply:
https://github.com/microsoft/TypeScript/issues/33932 for posterity